common/libs/units/units.h

```diff
@@ -363,4 +415,16 @@ constexpr inline Duration operator-(const Time &a, const Time &b) {
   return Duration(static_cast<int64_t>(a.val_ - b.val_));
 }
 
+constexpr inline Volume operator*(const VolumetricFlow &a, const Duration &b) {
+  return Volume(a.val_ * static_cast<float>(b.val_) / 1000.0f);
+}
+
+constexpr inline Volume operator*(const Duration &b, const VolumetricFlow &a) {
+  return Volume(a.val_ * static_cast<float>(b.val_) / 1000.0f);
+}
+
+constexpr inline VolumetricFlow operator/(const Volume &a, const Duration &b) {
+  return VolumetricFlow(a.val_ / static_cast<float>(b.val_) * 1000.0f);
```

Comment:
Instead of defining these in terms of `val_`, could we define them in terms of strongly-typed units?

```
constexpr inline Volume operator*(const VolumetricFlow &a, const Duration &b) {
  return liters(a.liters_per_second() * b.seconds());
}
```

This is a little safer because it's not relying on implementation details.  Also (I think) these also don't need to be friend functions, and you also also don't need the forward declares.

Reply:
OK, will try to figure these out tomorrow.

Reply:
Indeed, without reference to val_, it works like a charm without friend and without forward declare, but requires passing arguments without const & directive, which I guess is OK